detector.go

```diff
@@ -86,6 +87,14 @@ func (f *DetectorFactory) NewDetector(appDir, buildConfigDir, orderPath, platfor
 	if err := f.setOrder(detector, orderPath, logger); err != nil {
 		return nil, err
 	}
+
+	var amd platform.AnalyzedMetadata
```

Comment:
Does this need to be guarded by Platform API version? Is there always an analyzed toml available to read on every version?

Reply:
thanks - i think this is resolved by passing in the struct